### PR TITLE
Fix or disable warnings on latest VS dogfood

### PR DESCRIPTION
**Pull request by jkotas**

### CODE REVIEW

**Comment by jkotas** [comment]
cc @dotnet/jit-contrib

File: src/coreclr/jit/codegencommon.cpp [hunk 1]
```diff
@@ -11409,7 +11408,6 @@ void CodeGen::genRegCopy(GenTree* treeNode)
         //
         // There should never be any circular dependencies, and we will check that here.
 
-        GenTreeCopyOrReload* copyNode = treeNode->AsCopyOrReload();
```
Comment (jkotas) [inline src/coreclr/jit/codegencommon.cpp:11412]: If some of these locals exist to aid debugging, I am happy to revert. Just let me know...

File: src/coreclr/jit/lsra.cpp [hunk 2]
```diff
@@ -4217,7 +4202,7 @@ void LinearScan::unassignPhysReg(RegRecord* regRec, RefPosition* spillRefPositio
         return;
     }
 
-    regNumber victimAssignedReg = assignedInterval->physReg;
+    // regNumber victimAssignedReg = assignedInterval->physReg;
```
Comment (sandreenko) [inline src/coreclr/jit/lsra.cpp:4205]: should it be deleted instead?

File: eng/native/configurecompiler.cmake [hunk 3]
```diff
@@ -502,6 +502,13 @@ if (MSVC)
   add_compile_options($<$<COMPILE_LANGUAGE:C,CXX>:/Zc:inline>) # All inline functions must have their definition available in the current translation unit.
   add_compile_options($<$<COMPILE_LANGUAGE:C,CXX>:/Zc:forScope>) # Enforce standards-compliant for scope.
 
+  add_compile_options($<$<COMPILE_LANGUAGE:C,CXX>:/wd4065>)
```
Comment (sandreenko) [inline eng/native/configurecompiler.cmake:505]: what is the difference between this block and the block below `# Disable Warnings:`?

**Review by sandreenko — APPROVED** [review]
LGTM, thank you

File: src/coreclr/jit/lsra.cpp (hunk 2 above)
Comment (jkotas) [reply src/coreclr/jit/lsra.cpp:4205]: It is referenced in a TODO `#if 0` block below.

File: eng/native/configurecompiler.cmake (hunk 3 above)
Comment (jkotas) [reply eng/native/configurecompiler.cmake:505]: I do not know.

**Review by BruceForstall — APPROVED** [review]
Thanks!

IMO, it would be useful to comment for each `/wd` what the text of the warning is that is being disabled. Otherwise, it's just a big list and nobody goes to the docs to see why there are so many warnings disabled.

**Comment by jkotas** [comment]
> comment for each /wd what the text of the warning is that is being disabled

Done. Also, deleted a few `/wd` that do not seem to be necessary anymore.

**Review by stephentoub — APPROVED** [review]

**Comment by BruceForstall** [comment]
@jkotas fyi, the JIT has it's own set of disabling pragmas, https://github.com/dotnet/runtime/blob/705daa17c134f6485830785a1cd07a27c59261a1/src/coreclr/jit/jit.h#L28-L45. Probably don't need to touch these.

**Comment by jkotas** [comment]
>  JIT has it's own set of disabling pragmas,

The global warning disables kick for the JIT as well. I think the items in this list are either redundant with the global list or not needed anymore. I have tried to delete. Let's see what CI thinks.

**Comment by jkotas** [comment]
This change is affecting Windows only. All Windows CI legs passed.

**Pull request merged**